rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/cache/OracleTableMetaCache.java

```diff
@@ -147,19 +150,39 @@ private TableMeta resultSetMetaToSchema(DatabaseMetaData dbmd, String tableName)
 
                 }
             }
-
+            if (tm.getAllIndexes().isEmpty()) {
+                throw new ShouldNeverHappenException(String.format("Could not found any index in the table: %s", tableName));
+            }
+            List<String> pkcol = new ArrayList<>();
             while (rsPrimary.next()) {
                 String pkIndexName = rsPrimary.getString("PK_NAME");
                 if (tm.getAllIndexes().containsKey(pkIndexName)) {
                     IndexMeta index = tm.getAllIndexes().get(pkIndexName);
                     index.setIndextype(IndexType.PRIMARY);
+                } else {
+                    pkcol.add(rsPrimary.getString("COLUMN_NAME"));
                 }
             }
-            if (tm.getAllIndexes().isEmpty()) {
-                throw new ShouldNeverHappenException(String.format("Could not found any index in the table: %s", tableName));
+            if (!pkcol.isEmpty()) {
+                int times = 0;
+                for (Map.Entry<String, IndexMeta> entry : tm.getAllIndexes().entrySet()) {
+                    IndexMeta index = entry.getValue();
+                    if (index.getIndextype().value() == IndexType.UNIQUE.value()) {
+                        for (ColumnMeta col : index.getValues()) {
+                            if (pkcol.contains(col.getColumnName())) {
+                                times++;
```

Comment:
How about change it to Primary directly if pkcol constains col.getColumnName?

Reply:
> How about change it to Primary directly if pkcol constains col.getColumnName?
I couldn't agree with you more.   but i thought we need to check the situation that PK is multiple column PK (i was not sure if the seata will surpport  composite primary key)

Reply:
I have a idea that is when rsPrimary is not empty and put a new IndexMeta instead of change it. We separate the Unique Key and Primary Key. How do you think about this?

Reply:
We change the Unique Key to Primary key before because when we create the primary key and Oracle will create a unique key automatically but now look like the solution of change the key type is now that good.

Reply:
for oracle will create a unique key automatically, it's not happend in  all the situation. only ,if there is no unique index,it would create one automatically.but i thought my PR logic is right.if the index is unique and the column is exacly same as PK column.  it means the unique index is the PK.     I couldn't agree with you more.  it's a good idea that use a individual IndexMeta to indicate PK,actually,i had thought this solution before. but i didn't check all the code of the seata,i have no idea how many places need to update for.    IN some other Oracle like database, example, 达梦,the defalut unique index name that was created by when we created the PK always  was not the same as PK name.

Reply:
for oracle will create a unique key automatically, it's not happend in  all the situation. only ,if there is no unique index,it would create one automatically.but i thought my PR logic is right.if the index is unique and the column is exacly same as PK column.  it means the unique index is the PK.     I couldn't agree with you more.  it's a good idea that use a individual IndexMeta to indicate PK,actually,i had thought this solution before. but i didn't check all the code of the seata,i have no idea how many places need to update for.    IN some other Oracle like database, example, 达梦,the defalut unique index name that was created by when we created the PK always  was not the same as PK name.

Reply:
You do not need to change all the code in seata, just need to keep the number of primary key and index type is right that s ok. You can check out the main code here, it will no problem if you keep the logic as same as before.
```
io.seata.rm.datasource.sql.struct.TableMeta#getPrimaryKeyOnlyName
io.seata.rm.datasource.sql.struct.TableMeta#getPrimaryKeyMap
```

Reply:
I got it.I'll push a new commit later.

Reply:
I got it what you meaning is.but it's not easy to determine how can we get the indexMeta property.actually,the PK is constraint,the actual unique index was created by PK is stored in the view of all_indexes.it means we need to determine which unique index was a part of the PK(PK include constraint name and index name).
![image](https://user-images.githubusercontent.com/89766728/133022333-3cce4f84-ba91-47cd-a0dc-5dd00a87be52.png)
indeed,at step 2,it's not a indexname,it's a constraint name. maybe the current PR is good.    i thought the key  logic is accoding the PK constriant name to determine which unique index name was a part of the PK.

Reply:
OK. You idea was right. You still have to add some comments and let other developer understand . For example:
```
// when we create a primary key constraint oracle will uses and existing unique index.
// if we create a unique index before create a primary constraint in the same column will cause the problem that primary key constraint name was different from the unique name.
// a table can have only one primary key
List<String> pkcol = new ArrayList<>();

...
//save the columns that constraint primary key name was different from unique index name
pkcol.add(rsPrimary.getString("COLUMN_NAME"));

...
//find the index that belong to the primary key constraint
for (Map.Entry<String, IndexMeta> entry : tm.getAllIndexes().entrySet())
```

Reply:
thanks for your advice. I'll add some necessary comments later.